DOCS-1275: Add CLI module upload file, dir (#2052)
viamrobotics/docs: docs/modular-resources/upload/_index.md

viam module create --name <your-module-name> --org-id <your-org-id>
```

This command creates a new `meta.json` metadata file in your current working directory, which serves as a template.

If you later wish to make your module public, you can use the `viam module update` command to do so.

{{% /tab %}}
{{% tab name="Public" %}}

{{% alert title="Important" color="note" %}}

> Once you upload a module as public, you cannot change it back to private.

Reviewer: ```suggestion
Once you update your module to mark it as public, you cannot change it back to private.
```

???? (upload vs update confusion) (but I like the clarification)

Author: haha this is me trying to use command names as copy verbs. I actually change visibility with `update`. But you're absolutely right. Conceptually, this should be upload.

Reviewer: Ha! Maybe! (I agree with your comment before you edited it that the concept is indeed `upload`ing but the command is `update` )

Author: Haha right on. I think your copy change does convey this intricacy well!

viam module create --name <your-module-name> --public-namespace <your-unique-namespace>
```

This command creates a new `meta.json` metadata file in your current working directory, which serves as a template.

{{% /tab %}}
{{< /tabs >}}

3. Edit the newly-created `meta.json` file, and provide the required configuration information for your custom module by filling in the following fields.
   The `name` field is pre-populated using the `--name` you provided in the `viam module create` command, and `visibility` is set to `private` by default.

> 

Reviewer: [opt] Some people have modified their `module_id` field without understanding that it is what uniquely identifies their module (and then they file confusing bug reports about missing modules when they try to update their module).

Might be worth it to write a short comment saying that module id should not be changed. Up to you

> 

Reviewer 2: ```suggestion

{{< alert title="Caution" color="caution" >}}
The `module_id` uniquely identifies your module. Do not change the `module_id`.
{{< /alert >}}

```

   On a successful update, the command will return a link to the updated module in the Viam registry.

5. Package your custom module to get it ready to upload to the Viam registry.
   Currently, the Registry only supports `tar.gz` or `tar.xz` format.
   Use the command below specific for the language of your module:

   - To package a module written in Go, run the following commands from the same directory as your `meta.json` file:

     ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
     go build -o bin/module ./module/main.go
     tar -czf module.tar.gz bin/module
     ```

     For more information, see [Compile a module into an executable](/modular-resources/create/#compile-the-module-into-an-executable).

   - To package a module written in Python, run the following command from the same directory as your `meta.json` file:

     ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
     tar -czf module.tar.gz run.sh requirements.txt src

>      ```

Reviewer: We also dont want python users to necessarily zip up their whole module directory because they probably have things like their python virtual environments in the directory (which will cause issues on other computers)

>      ```

Author: Great point, this makes perfect sense. Retaining portions of this, in Python context only. Thank you for the helpful comment!!
